docker-compose.yml

```diff
@@ -8,93 +8,144 @@ services:
-    command: bash -c "/htapps/babel/imgsrv/bin/startup_imgsrv"
+      - "41028:41028"
+
+  vufind:
+    build: ./catalog
+    hostname: 'catalog-dev'
+    volumes:
+      - ./catalog:/app
+    depends_on:
+      - mysql-sdr
+      - solr-sdr-catalog
+
+#### DATA STORES
 
   solr-sdr-catalog:
     image: ghcr.io/hathitrust/catalog-solr-sample
     ports:
       - "9033:9033"
 
+  solr-lss-dev:
+    image: solr:6
+    ports:
+      - "8983:8983"
+    user: ${CURRENT_USER}
+    volumes:
+      - ./lss_solr_configs/lss-dev/core-x:/opt/solr/server/solr/core-x
+      - ./lss_solr_configs/lss-dev/core-y:/opt/solr/server/solr/core-y
+      - ./lss_solr_configs:/opt/lss_solr_configs
+      - ./lss_solr_configs/lib:/opt/solr/server/solr/lib
+      - ./logs/solr:/opt/solr/server/logs
+
   mysql-sdr:
-    image: mariadb
+    image: ghcr.io/hathitrust/db-image:latest
     volumes:
-      - ../catalog/docker/vufind.sql:/docker-entrypoint-initdb.d/vufind.sql
-      - ../imgsrv/vendor/common-lib/lib/sql/000_ht_schema.sql:/docker-entrypoint-initdb.d/0000_ht_schema.sql
-      - ../imgsrv/vendor/common-lib/lib/sql/001_ht_ht_namespaces.sql:/docker-entrypoint-initdb.d/001_ht_ht_namespaces.sql
-      - ../imgsrv/vendor/common-lib/lib/sql/002_ht_rights_current.sql:/docker-entrypoint-initdb.d/002_ht_rights_current.sql
-      - ../imgsrv/sql/100_ht_web_schema.sql:/docker-entrypoint-initdb.d/100_ht_web_schema.sql
-      - ../imgsrv/sql/200_users.sql:/docker-entrypoint-initdb.d/200_users.sql
+      - ./slip/etc/sql/100_slip.sql:/docker-entrypoint-initdb.d/100_slip.sql
+      - ./catalog/docker/vufind.sql:/docker-entrypoint-initdb.d/101_vufind.sql
+    ports:
+      - "3307:3306"
 
     environment:
-      # - MARIADB_RANDOM_ROOT_PASSWORD=1
+      # - mysql-sdr_RANDOM_ROOT_PASSWORD=1
       MYSQL_ROOT_PASSWORD: TIMTOWTDIBSCINABTE
 
-  vufind:
-    build: ../catalog
-    hostname: 'catalog-dev'
+#### INDEXING
+
+  # We add the 'indexing' profile to keep these from starting automatically
+  # when we do 'docker-compose up'
+
+  slip:
+    build: ./slip
+    image: hathitrust-slip
+    user: ${CURRENT_USER}
     volumes:
-      - ../catalog:/app
+      - ./slip:/htapps/babel/slip
+      - ./logs:/htapps/babel/logs
+      # this is where docs-j saves output
+      - ./slip/sample:/htapps/babel/logs/tmp
+      - "./sample-data/sdr1:/sdr1"
+      - "./sample-data/etc:/htapps/babel/etc"
+      - "./sample-data:/tmp/sample-data"
+    environment:
+      - SDRROOT=/htapps/babel
+      - SDRDATAROOT=/sdr1
+      - HT_DEV=docker
     depends_on:
       - mysql-sdr
       - solr-sdr-catalog
+    command: bash
+    profiles:
+      - indexing
+
+  traject:
+    build: ./hathitrust_catalog_indexer
+    user: ${CURRENT_USER}
+    environment:
+      - SOLR_URL=http://solr-sdr-catalog:9033/solr/catalog
+      - redirect_file=/dev/null
+      - NO_DB=1
+      - DDIR=/app/metadata
+    depends_on:
+      - solr-sdr-catalog
+    volumes:
+      - "./sample-data/metadata:/app/metadata"
```

Comment:
Needs `- ./hathitrust_catalog_indexer:/app` for the script to be found in `stage_item.rb`.

Reply:
But then I get 

```
bundler: command not found: traject
Install missing gem executables with `bundle install`
```

It doesn't seem any `bundle install` persists between `babel_traject_run` calls.

Reply:
Argh. I think this is a side effect of using the `hathitrust_catalog_indexer` via the checked out copy rather than via the image. I will make the required changes...

Reply:
This should be fixed now.